ui/apps/platform/src/services/GroupsService.js

```diff
@@ -23,6 +23,23 @@ export function updateOrAddGroup({ oldGroups, newGroups }) {
     return axios.post(updateUrl, { previous_groups: oldGroups, required_groups: newGroups });
 }
 
+/**
+ * Fetches the default rule.
+ *
+ * @returns {Promise<Object, Error>} fulfilled default group
+ */
+export function getDefaultGroup({ authProviderId, roleName }) {
+    // The default group is characterized by the following:
+    // - Only authProviderID is set, key and value are empty.
+    // - The role name of the group matches the given role name.
+    // We need _explicitly_ ask for empty key and value fields to receive the actual default role.
+    return axios
+        .get(`${url}?authProviderId=${authProviderId}&key=&value=&roleName=${roleName}`)
+        .then((response) => ({
+            response: response.data?.groups[0],
```

Comment:
Does this mean that if _multiple_ default groups have been created (should be possible now from the backend perspective, e.g., via `curl`), a batch group request will delete all except the one that we stored here? Looks like a bug.

Reply:
@rukletsov Can you elaborate on what you now expect for multiple default groups?

Just because it is technically possible through the API, is it a requirement, or even an option, that we need to expose to customers?

The current UI has one form input, labelled "Minimum access role" that only allows you select one role from a list. Has this requirement changed?

Reply:
@rukletsov 
While I agree it's technically possible to do so, I feel like we should be more strict on the datastore regarding this ambiguity.

I don't think we are ready to change requirements in the UI for this, i.e. allowing multiple "Minimum access roles" to be set (which would turn out to be multiple default groups) nor do I personally want this, from a user point of view it would be confusing.

Instead, if we decide to do anything, I feel we should restrict the API. **Iff** the client creates a group with only the auth provider ID set, it's not possible to do so when another group with only the auth provider ID set exists. This way, we wouldn't have to deal with the ambiguity in the UI, can keep it as-is, and don't have to worry about potentially removing default groups here. However, this would mean we restrict the behavior from the API.

Otherwise, if we decide to change anything in the UI, how would you even know which default group to change? Let's say you have multiple groups with the same role name (which is what you would yield here): which group is the one originally created, and which is the one not and shouldn't be changed?

Reply:
If you will forgive a question out of an abundance of ignorance, is there any situation in which `groups` might be an empty array? Or more relevant, any situation which a user can cause to happen?

Apparently yes, because `getGroupsWithDefault` function has conditional logic for the situation:

```js
// The default group is not yet created, meaning we have to make sure we create it here.
// Set the auth provider ID and role name.
// Use the default group if we receive a value here.
```

Reply:
This is one of the most important files in services folder that is still JavaScript. Adding types/group.proto.ts and rewriting this file in TypeScript might be a good collaborative follow up for UI and Merlin in 72.

Reply:
@pedrottimark Yes, this may happen.
It could be, as you rightly mentioned, the auth provider is currently created and the default group has not been created yet _or_ the default group has been deleted via API (which is unlikely, but still possible).

I agree on the collaborative effort to migrate this to typescript, it would certainly make life better. I've created [ROX-11600](https://issues.redhat.com/browse/ROX-11600) for that with the aim to complete it within 3.72.

Reply:
@vjwilson @dhaus67 Let's restrict the API to at most one default rule / group per auth provider. @dhaus67 could you please send a new PR in this chain for this?